projects/prometheus/alertmanager/Makefile

```diff
@@ -0,0 +1,79 @@
+REPO?=alertmanager
+CLONE_URL?=https://github.com/prometheus/$(REPO).git
+GIT_COMMIT?=$(shell cat GIT_COMMIT)
+
+ifeq ("$(CLONE_URL)","")
+	$(error No clone url was provided.)
+endif
+
+ifeq ("$(REPO)","")
+	$(error No repository name was provided.)
+endif
+
+ifeq ("$(GIT_COMMIT)","")
+	$(error No git commit was provided.)
+endif
+
+DEVELOPMENT?=true
+AWS_ACCOUNT_ID?=$(shell aws sts get-caller-identity --query Account --output text)
+AWS_REGION?=us-west-2
+
+IMAGE_ARCH?="amd64"
+IMAGE_OS?="linux"
+IMAGE_REPO?=$(AWS_ACCOUNT_ID).dkr.ecr.$(AWS_REGION).amazonaws.com
+IMAGE_NAME?=prometheus/alertmanager
+# This tag is overwritten in the prow job to point to the commit hash
+IMAGE_TAG?=latest
+IMAGE?=$(IMAGE_REPO)/$(IMAGE_NAME):$(IMAGE_TAG)
+
+.PHONY: clone
+clone: clean
+	git clone $(CLONE_URL) $(REPO)
+	cd $(REPO) && git checkout $(GIT_COMMIT)
+
+.PHONY: binaries
+binaries:
+	make build -C $(REPO)
+
+.PHONY: move-binaries
+move-binaries:
+	mkdir -p $(REPO)/.build/$(IMAGE_OS)-$(IMAGE_ARCH)
```

Comment:
Instead of defining the image_arch and image_os above, can we copy the all the tools to the `.build/` and reference the appropriate binary using container args in our Dockerfile? That way, if a customer were to run it for linux arm64, there wouldn't be additional changes needed to this Makefile.

Reply:
The upstream Dockerfile expects the tools to be in the `.build/OS-ARCH` directory and they have used ARGS in the Dockerfile to pass the OS and ARCH. If we change that, we will need to maintain a separate Dockerfile. https://github.com/prometheus/alertmanager/blob/master/Dockerfile